src/App.tsx

```diff
@@ -10,33 +11,27 @@ export const tabs = [
 ];
 
 export const App: React.FC = () => {
+  const [selectedTab, setSelectedTab] = useState(tabs[0]);
+
+  const onTabSelected = (tab: Tab) => {
+    const index = tabs.findIndex(t => t.id === tab.id);
+
+    setSelectedTab(tabs[index !== -1
```

Comment:
better to calculate it outside. The code will be larger, but much better in terms of redability